src/Compiler/Checking/Expressions/CheckComputationExpressions.fs

```diff
@@ -1900,105 +1900,162 @@ let rec TryTranslateComputationExpression
-                        mBind
-                    )
+                    let rhsExpr =
+                        mkSourceExprConditional isFromSource rhsExpr ceenv.sourceMethInfo ceenv.builderValName
 
-                let rhsExpr =
-                    mkSourceExprConditional isFromSource rhsExpr ceenv.sourceMethInfo ceenv.builderValName
+                    mkSynCall "Bind" mBind [ rhsExpr; consumeExpr ] ceenv.builderValName
+                    |> addBindDebugPoint spBind
 
-                mkSynCall "Bind" mBind [ rhsExpr; consumeExpr ] ceenv.builderValName
-                |> addBindDebugPoint spBind
+                Some(translatedCtxt bindExpr)
 
-            Some(translatedCtxt bindExpr)
+            | SynPat.Wild _ when cenv.g.langVersion.SupportsFeature(LanguageFeature.UseBangBindingValueDiscard) ->
+                if not (isNil andBangs) then
+                    let m =
+                        match andBangs with
+                        | [] -> comp.Range
+                        | h :: _ -> h.Trivia.AndBangKeyword
 
-        // 'use! pat = e1 ... in e2' where 'pat' is not a simple name -> error
-        | SynExpr.LetOrUseBang(isUse = true; andBangs = andBangs; trivia = { LetOrUseBangKeyword = mBind }) ->
-            if isNil andBangs then
-                error (Error(FSComp.SR.tcInvalidUseBangBinding (), mBind))
-            else
-                let m =
-                    match andBangs with
-                    | [] -> comp.Range
-                    | h :: _ -> h.Trivia.AndBangKeyword
+                    error (Error(FSComp.SR.tcInvalidUseBangBindingNoAndBangs (), m))
+                else
+                    if ceenv.isQuery then
+                        error (Error(FSComp.SR.tcBindMayNotBeUsedInQueries (), mBind))
+
+                    if
+                        isNil (
+                            TryFindIntrinsicOrExtensionMethInfo
+                                ResultCollectionSettings.AtMostOneResult
+                                cenv
+                                ceenv.env
+                                mBind
+                                ceenv.ad
+                                "Using"
+                                ceenv.builderTy
+                        )
+                    then
+                        error (Error(FSComp.SR.tcRequireBuilderMethod "Using", mBind))
+
+                    if
+                        isNil (
+                            TryFindIntrinsicOrExtensionMethInfo
+                                ResultCollectionSettings.AtMostOneResult
+                                cenv
+                                ceenv.env
+                                mBind
+                                ceenv.ad
+                                "Bind"
+                                ceenv.builderTy
+                        )
+                    then
+                        error (Error(FSComp.SR.tcRequireBuilderMethod "Bind", mBind))
+
```

Comment:
Is it possible to also deduplicate this block with the one for `SynPat.Named` and `SynPat.LongIdent` above?
It appears to be using the same values and calls and errors... ?

Reply:
Done